tests/roles/dataplane_adoption/tasks/main.yaml

```diff
@@ -14,6 +14,37 @@
     $(cat {{ edpm_privatekey_path }} | base64 | sed 's/^/        /')
     EOF
 
+- name: stop libvirt services
+  no_log: "{{ use_no_log }}"
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+
+    ServicesToStop=("tripleo_nova_api_cron.service"
+                    "tripleo_nova_api.service"
+                    "tripleo_nova_compute.service"
+                    "tripleo_nova_conductor.service"
+                    "tripleo_nova_libvirt.target"
+                    "tripleo_nova_metadata.service"
+                    "tripleo_nova_migration_target.service"
+                    "tripleo_nova_scheduler.service"
+                    "tripleo_nova_virtlogd_wrapper.service"
+                    "tripleo_nova_virtnodedevd.service"
+                    "tripleo_nova_virtproxyd.service"
+                    "tripleo_nova_virtqemud.service"
+                    "tripleo_nova_virtsecretd.service"
+                    "tripleo_nova_virtstoraged.service"
+                    "tripleo_nova_vnc_proxy.service")
+
+    echo "Stopping nova services"
+
+    for service in ${ServicesToStop[*]}; do
+        echo "Stopping the service: $service in each controller node"
+        $CONTROLLER1_SSH sudo systemctl stop $service
+        $CONTROLLER2_SSH sudo systemctl stop $service
+        $CONTROLLER3_SSH sudo systemctl stop $service
+    done
```

Comment:
The shell variables will be undefined here. We should do something like https://github.com/openstack-k8s-operators/data-plane-adoption/blob/main/tests/roles/mariadb_copy/tasks/main.yaml#L8-L28